src/shogun/neuralnets/Autoencoder.h

```diff
@@ -165,6 +165,44 @@ class CAutoencoder : public CNeuralNetwork
 
 	virtual const char* get_name() const { return "Autoencoder"; }
 
+	/** Sets noise type for denoising autoencoders.
+	 *
+	 * If set to AENT_DROPOUT, inputs are randomly set to zero during each
+	 * iteration of training with probability noise_parameter.
+	 *
+	 * If set to AENT_GAUSSIAN, gaussian noise with zero mean and noise_parameter
+	 * standard deviation is added to the inputs.
+	 *
+	 * Default value is AENT_NONE
+	 * @param _noise_type noise type for denoising autoencoders
+	 */
+	void set_noise_type(EAENoiseType _noise_type)
+	{
+		this->noise_type = _noise_type;
```

Comment:
mmh.
Usually, members in shogun have an "m_" prefix, which avoid the need for underscores in setter variables....
But well, its ok for now
@lisitsyn